File: 100 days of leetcode/Day1.cpp
// class Solution {
// public:
//     vector<int> twoSum(vector<int>& nums, int target) {

//         // Good for memory
//         vector<int> list;
//         for(int i = 0; i < nums.size(); i++){
//             for(int j = i+1; j < nums.size(); j++ ){
//                 if(nums[i] + nums[j] == target){
//                     list.push_back(i);
//                     list.push_back(j);
//                     // return {nums[i], nums[j]};
//                 }
//             }
//         }
//         return list;
//     }
// };


// Good for time complexity
#include <vector>
#include <unordered_map>

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> numToIndex;
        vector<int> result;
        
        for (int i = 0; i < nums.size(); i++) {
            int complement = target - nums[i];
            if (numToIndex.find(complement) != numToIndex.end()) {
                result.push_back(numToIndex[complement]);
                result.push_back(i);
                break;
            }
            numToIndex[nums[i]] = i;
        }
        
        return result;
    }
};

File: 100 days of leetcode/Day1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        #Memory efficient
        for i in range(0, len(nums)-1):
            for j in range(i+1, len(nums)):
                if (nums[i] + nums[j] == target):
                    return [i, j]

        #Runtime efficient
        # seen = {}
        # for i, num in enumerate(nums):
        #     remaining = target - num
            
        #     if remaining in seen:
        #         return [i, seen[remaining]]
            
        #     seen[num] = i

\
